src/libDirectoryService/PoW2Processing.cpp

```diff
@@ -197,37 +197,25 @@ bool DirectoryService::ProcessPoW2Submission(
 #ifndef IS_LOOKUP_NODE
     // Message = [32-byte block num] [4-byte listening port] [33-byte public key] [8-byte nonce] [32-byte resulting hash] [32-byte mixhash]
     LOG_MARKER();
-    // shared_lock<shared_timed_mutex> lock(m_mutexProducerConsumer);
-    unsigned int sleep_time_while_waiting = 100;
+
     if (m_state == DSBLOCK_CONSENSUS
         || (m_state != POW2_SUBMISSION && m_mode == Mode::IDLE
             && m_mediator.m_node->m_state == Node::POW2_SUBMISSION))
     {
-        for (unsigned int i = 0; i < POW_SUB_BUFFER_TIME; i++)
+        std::unique_lock<std::mutex> cv_lk(m_MutexCVPOW2Submission);
+
+        if (cv_POW2Submission.wait_for(
+                cv_lk, std::chrono::seconds(POW_SUBMISSION_TIMEOUT))
+            == std::cv_status::timeout)
         {
-            if (m_state == POW2_SUBMISSION)
-            {
-                break;
-            }
-            if (i % 100 == 0)
-            {
-                LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-                          "Waiting for POW2_SUBMISSION state before "
-                          "processing. Current state is "
-                              << m_state);
-            }
-
-            // Magic number for now.
-            // Basically, I want to wait for awhile for the dsblock to arrive before
-            // I request for one.
-            if (!m_requesting_last_ds_block and i == 300)
-            {
-                m_requesting_last_ds_block = true;
-                LastDSBlockRequest();
-            }
-            this_thread::sleep_for(
-                chrono::milliseconds(sleep_time_while_waiting));
+            LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+                      "Time out while waiting for state transition and "
+                      "consensus object creation ");
         }
+
+        LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                  "State transition is completed and consensus object "
+                  "creation. (check for timeout)");
```

Comment:
Can this log message be rephrased a bit? It doesn't read well now. Others similar ones also need some change.